chain/listener/listener.go

```diff
@@ -119,6 +119,27 @@ func (l *listener) pollBlocks() error {
 				continue
 			}
 
+			// debugging
+
+			blockData, err := l.client.BlockByNumber(context.Background(), latestBlock)
+			if err != nil {
+				return err
+			}
+
+			// fetch IstanbulExtra data by parsing block header
+			// https://github.com/celo-org/celo-blockchain/blob/master/core/types/istanbul.go#L128-L142
+			extra, err := types.ExtractIstanbulExtra(blockData.Header())
+			if err != nil {
+				return err
+			}
+
+			apk, err := l.valsAggr.GetAPKForBlock(latestBlock, uint8(l.cfg.ID), l.cfg.EpochSize, extra)
+			if err != nil {
+				return err
+			}
+
+			log.Debug().Msgf("APK: %x", apk)
```

Comment:
this looks totally useless since you are doing same manipulations in  getDepositEventsAndProofsForBlock subroutine

Reply:
yes, it was just there for debugging

<img width="517" alt="Screen Shot 2021-06-22 at 2 51 23 PM" src="https://user-images.githubusercontent.com/17755587/123005151-81a4fe00-d36a-11eb-9cea-c9b96af5ee58.png">